### PR TITLE
Add config option to disable 404 page

### LINKED ISSUE #1080: Subpath: This route collides with: "src/pages/404.astro".
### What version of `starlight` are you using?

0.12.1

### What version of `astro` are you using?

3.3.0

### What package manager are you using?

yarn

### What operating system are you using?

Mac

### What browser are you using?

Chrome

### Describe the Bug

I already used Astro for my website and wanted to integrate Astro Starlight manually  to it for my "Help Desk". When I tried to start it, I got the following error:

```
This route collides with: "src/pages/404.astro".
```

Because Astro Starlight comes with a 404 as well, I had to remove my top-level 404 page. It works, however, are there any plans to allow two 404 if Starlight gets dropped in as a [subpath](https://starlight.astro.build/manual-setup/#tips-for-existing-sites)?

Thanks for all your work here!

### Link to Minimal Reproducible Example

-

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by connor-baer**
#### Description

- Closes #1080
- Add a new user config option `disable404Route` to disable injecting Starlight's 404 page
- Update the configuration reference with the new option
- TODO: Update the [“Custom 404 page”](https://starlight.astro.build/guides/customization/#custom-404-page) section in the docs

### CODE REVIEW

File: docs/src/content/docs/reference/configuration.mdx [hunk 1]
```diff
@@ -514,6 +514,13 @@ Sets the delimiter between page title and site title in the page’s `<title>` t
 By default, every page has a `<title>` of `Page Title | Site Title`.
 For example, this page is titled “Configuration Reference” and this site is titled “Starlight”, so the `<title>` for this page is “Configuration Reference | Starlight”.
 
+### `disable404Route`
+
+**type:** `boolean`  
+**default:** `'false'`
+
+Disables injecting Starlight's default [404 page](https://docs.astro.build/en/core-concepts/astro-pages/#custom-404-error-page) so it can be overwritten with a custom one.
```
Comment (delucis) [inline docs/src/content/docs/reference/configuration.mdx:522]: Suggesting a minor tweak here to make the second part more “user desire => action”.

```suggestion
Disables injecting Starlight's default [404 page](https://docs.astro.build/en/core-concepts/astro-pages/#custom-404-error-page). To use a custom `src/pages/404.astro` route in your project, set this option to `false`.
```

**Review by delucis — COMMENTED** [review]
Awesome work @connor-baer! This looks great to me even without the extra todo item — we can always tackle that later. Left one small suggestion on the docs, but the code seems spot on 🙌

**Review by delucis — APPROVED** [review]
Thanks again for your contribution @connor-baer! Hope you don’t mind I jumped in to polish up the last docs changes so we can get this released today.

Appreciate you helping out 🌟

**Comment by connor-baer** [comment]
@delucis, I don't mind at all, thanks for finishing this up! 🙌🏻

**Pull request merged**